src/DotLiquid/StandardFilters.cs

```diff
@@ -931,6 +931,53 @@ public static IEnumerable Compact(object input)
             ary.RemoveAll(item => item == null);
             return ary;
         }
+
+        /// <summary>
+        /// Creates an array including only the objects with a given property value, or any truthy value by default.
+        /// </summary>
+        /// <param name="input">an array to be filtered</param>
+        /// <param name="property">The name of the property to filter by</param>
+        /// <param name="target_value">Value to retain, if null object containing this property are retained</param>
+        /// <returns></returns>
+        public static IEnumerable Where(object input, string property, string target_value = null)
+        {
+            if (input == null)
+                return null;
+
+            if (property.IsNullOrWhiteSpace())
+                throw new ArgumentNullException(paramName: nameof(property), message: $"'{nameof(property)}' cannot be null or empty.");
+
+            List<object> inputList = (input is IEnumerable enumerableInput) ? enumerableInput.Cast<object>().ToList() : null;
+
+            return inputList.Where(s => s.Matches(property, target_value));
+        }
+
+        private static bool Matches(this object any, string property, string target_value)

```

Comment:
This matches helper method implements the specific Shopify liquid concept of Where filter match. See if there is an appropriate method name to describe that purpose. Either way, do XML comment this method to explain what it is going and what are each of the input parameters.

Reply:
Renamed to HasMatchingProperty() and added comment.